crates/extensions/c8y_mapper_ext/src/tests.rs

```diff
@@ -3113,6 +3115,52 @@ async fn mapper_converts_config_cmd_to_supported_op_and_types_for_child_device()
     .await;
 }
 
+#[tokio::test]
+async fn mapper_publishes_all_supported_operations_on_signal() {
+    let ttd = TempTedgeDir::new();
+    ttd.dir("operations").dir("c8y").file("c8y_Restart");
+    ttd.dir("operations")
+        .dir("c8y")
+        .dir("test::device::child01")
+        .file("c8y_Restart");
+    ttd.dir("operations")
+        .dir("c8y")
+        .dir("test::device::child02")
+        .file("c8y_Restart");
+
+    let test_handle = spawn_c8y_mapper_actor(&ttd, true).await;
+    let TestHandle { mqtt, .. } = test_handle;
+    let mut mqtt = mqtt.with_timeout(TEST_TIMEOUT_MS);
+
+    skip_init_messages(&mut mqtt).await;
+
+    // Register the tedge-mapper-c8y service upfront
+    mqtt.send(MqttMessage::new(
+        &Topic::new_unchecked("te/device/main/service/tedge-mapper-c8y"),
+        r#"{"@parent":"device/main//","@type":"service","name":"tedge-mapper-c8y","type":"service"}"#,
+    ))
+    .await
+    .expect("Send failed");
+    mqtt.skip(1).await; // Skip 102 message
+
+    mqtt.send(MqttMessage::new(
+        &Topic::new_unchecked("te/device/main/service/tedge-mapper-c8y/signal/sync_operations"),
+        "{}",
+    ))
+    .await
+    .expect("Send failed");
+
+    assert_received_contains_str(
+        &mut mqtt,
+        [
+            ("c8y/s/us", "114,c8y_Restart"),
+            ("c8y/s/us/test::device::child01", "114,c8y_Restart"),
+            ("c8y/s/us/test::device::child02", "114,c8y_Restart"),
```

Comment:
To have a test that is more representative, I would a child device with no operations and another one with at least 2 operations.

Reply:
Fixed in 03491b935c35aef7b0be070d7a92f66b52c9d14c

Reply:
Indeed fixed by https://github.com/thin-edge/thin-edge.io/pull/3761/commits/0525623ba7fbcc11775ca3552ed7a1329c3bcc64